pkcs7/benchmark-streaming-envelop.c

```diff
@@ -0,0 +1,412 @@
+/* benchmark-streaming-envelop.c
+ *
+ * Copyright (C) 2006-2025 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL. (formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include <wolfssl/options.h>
+#include <wolfssl/wolfcrypt/pkcs7.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/wc_port.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+#define USE_CERT_BUFFERS_2048
+#include <wolfssl/certs_test.h>
+
+#include <stdio.h>
+#include <sys/time.h>
+
+#define CONTENT_FILE_NAME "benchmark-content.bin"
+#define ENCODED_FILE_NAME "test-stream-dec.p7b"
+#define DECODED_FILE_NAME "benchmark-decrypted.bin"
+#define TEST_STREAM_CHUNK_SIZE 1000
```

Comment:
Does changing the chunk size affect performance significantly?
If so, then I would like it to be a command line parameter.

Reply:
Changing the size does, I found a bug when testing different sizes. Working on resolving it.

Reply:
Added a pull request to wolfSSL for the bug found (https://github.com/wolfSSL/wolfssl/pull/8634)